packages/react-component-library/src/components/DatePickerE/useInput.ts

```diff
@@ -35,40 +34,42 @@ export function useInput(
         return
       }
 
-      if (isValid(from)) {
-        setInputValue(formatDatesForInput(from, null, datePickerFormat))
-      }
+      dispatch({ type: 'refreshInputValue' })
     },
-    [datePickerFormat, from, isRange, setInputValue]
+    [dispatch, isRange]
   )
 
   const handleInputBlur = useCallback(() => {
     if (isRange) {
       return
     }
 
-    if (isValid(from)) {
-      setInputValue(formatDatesForInput(from, null, datePickerFormat))
-    }
+    dispatch({ type: 'refreshInputValue' })
 
     setHasError(from && !isValid(from))
-  }, [datePickerFormat, from, isRange, setHasError, setInputValue])
+  }, [dispatch, from, isRange, setHasError])
 
   const handleInputChange = useCallback(
     (event) => {
       if (isRange) {
         return
       }
 
-      setInputValue(event.currentTarget.value)
+      dispatch({
+        type: 'update',
```

Comment:
Is it worth using constants for these action types? See Timeline for an example.

Reply:
I'm not 100% convinced it's an improvement. But have added for consistency :smile: